src/keeshare/KeeShareSettings.cpp

```diff
@@ -157,15 +116,29 @@ namespace KeeShareSettings
             if (reader.name() == "Signer") {
                 certificate.signer = reader.readElementText();
             } else if (reader.name() == "Key") {
-                certificate.key = QByteArray::fromBase64(reader.readElementText().toLatin1());
+                auto rawKey = QByteArray::fromBase64(reader.readElementText().toLatin1());
+                if (!rawKey.isEmpty()) {
+                    try {
+                        Botan::DataSource_Memory dataSource(reinterpret_cast<const uint8_t*>(rawKey.constData()),
+                                                            rawKey.size());
+                        certificate.key.reset(Botan::PKCS8::load_key(dataSource).release());
+                    } catch (std::exception& e) {
+                        qWarning("KeeShare: Failed to deserialize key data: %s", e.what());
```

Comment:
This deserves proper error handling.

Reply:
We are going to be removing all of this code in the near future by removing signed keeshare container support. It is totally unnecessary. Recommend just keeping it as-is for this PR.